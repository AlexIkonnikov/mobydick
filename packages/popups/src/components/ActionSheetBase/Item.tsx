import {TouchableOpacity, View} from '@npm/mobydick-core';
import {ViewStyle} from 'react-native';
import React, {FC, ReactElement} from 'react';
import {useStyles} from '@npm/mobydick-styles';
import {TypographyProp} from '@npm/mobydick-typography';

import stylesCreate from './stylesCreate';
import getContents from './content/getContents';

export interface IPropsItem {
  title?: string;
  subTitle?: string;
  label?: string;
  labelFont?: TypographyProp;
  onPress?(): void;
  style?: ViewStyle;
  textFont?: TypographyProp;
  leftIcon?: ReactElement;
  selected?: string[];
  disabled?: boolean;
}

const Item: FC<IPropsItem> = props => {
  const {
    title,
    subTitle,
    textFont,
    onPress,
    style,
    leftIcon,
    selected,
    disabled,
  } = props;
  const [styles] = useStyles(
    stylesCreate,
    Boolean(selected),
    Boolean(leftIcon),
  );

  if (disabled) {
    return (
      <View style={[styles.item, style]}>
        {getContents(leftIcon, subTitle, selected, textFont, title)}
      </View>
    );
  }

  return (
    <TouchableOpacity
      activeOpacity={0.9}
      style={[styles.item, style]}
      onPress={onPress}>
      {getContents(leftIcon, subTitle, selected, textFont, title)}
    </TouchableOpacity>
  );
};

export default Item;
