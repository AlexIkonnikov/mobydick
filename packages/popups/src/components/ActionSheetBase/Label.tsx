import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';
import {Typography, TypographyProp} from '@npm/mobydick-typography';
import {ViewStyle} from 'react-native';
import {View} from '@npm/mobydick-core';

import stylesCreate from './stylesCreate';

export interface IPropsLabel {
  label: string;
  labelFont?: TypographyProp;
  style?: ViewStyle;
}

const Label: FC<IPropsLabel> = props => {
  const [styles] = useStyles(stylesCreate);
  const {label, labelFont, style} = props;

  return (
    <View style={[styles.item, style]}>
      <Typography
        style={styles.label}
        font={labelFont ? labelFont : 'Regular-Tertiary-XXS'}>
        {label}
      </Typography>
    </View>
  );
};

export default Label;
