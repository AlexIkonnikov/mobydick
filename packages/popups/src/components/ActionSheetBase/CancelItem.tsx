import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';

import stylesCreate from './stylesCreate';
import {IPropsItem} from './Item';

import {ActionSheetBase} from './index';

const CancelItem: FC<IPropsItem> = props => {
  const [styles] = useStyles(stylesCreate);
  const {...otherProps} = props;
  return <ActionSheetBase.Item style={styles.cancelItem} {...otherProps} />;
};

export default CancelItem;
