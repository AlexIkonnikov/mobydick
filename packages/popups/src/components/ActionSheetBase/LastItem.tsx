import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';

import stylesCreate from './stylesCreate';
import Item, {IPropsItem} from './Item';

const LastItem: FC<IPropsItem> = props => {
  const [styles] = useStyles(stylesCreate);

  return <Item style={styles.lastItem} {...props} />;
};

export default LastItem;
