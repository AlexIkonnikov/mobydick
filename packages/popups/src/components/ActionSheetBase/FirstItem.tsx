import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';

import stylesCreate from './stylesCreate';
import Item, {IPropsItem} from './Item';
import Label, {IPropsLabel} from './Label';

const FirstItem: FC<IPropsItem | IPropsLabel> = props => {
  const [styles] = useStyles(stylesCreate);
  const {label, ...otherProps} = props;
  return label ? (
    <Label style={styles.firstItem} label={label} {...otherProps} />
  ) : (
    <Item style={styles.firstItem} {...otherProps} />
  );
};

export default FirstItem;
