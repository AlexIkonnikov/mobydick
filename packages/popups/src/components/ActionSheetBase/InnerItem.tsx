import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';

import stylesCreate from './stylesCreate';
import Item, {IPropsItem} from './Item';
import Label, {IPropsLabel} from './Label';

const InnerItem: FC<IPropsItem | IPropsLabel> = props => {
  const [styles] = useStyles(stylesCreate);
  const {label, ...otherProps} = props;
  return label ? (
    <Label style={styles.innerItem} label={label} {...otherProps} />
  ) : (
    <Item style={styles.innerItem} {...otherProps} />
  );
};

export default InnerItem;
