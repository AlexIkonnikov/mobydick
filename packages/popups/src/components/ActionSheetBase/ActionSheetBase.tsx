import {View} from '@npm/mobydick-core';
import React, {FC} from 'react';
import {StyleProp, ViewStyle} from 'react-native';

import {IPopup} from '../../types';
import {PopupBase} from '../PopupBase';

import Item from './Item';
import FirstItem from './FirstItem';
import InnerItem from './InnerItem';
import LastItem from './LastItem';
import CancelItem from './CancelItem';
import Label from './Label';

const ActionSheetBase: FC<
  Omit<IPopup, 'Content'> & {
    onClose: () => void;
    containerStyle?: StyleProp<ViewStyle>;
  }
> & {
  Item: typeof Item;
  FirstItem: typeof FirstItem;
  InnerItem: typeof InnerItem;
  LastItem: typeof LastItem;
  CancelItem: typeof CancelItem;
  Label: typeof Label;
} = props => {
  const {children, overlayStyle, onClose, containerStyle} = props;

  return (
    <PopupBase onClose={onClose} overlayStyle={overlayStyle}>
      <View style={[containerStyle]}>{children}</View>
    </PopupBase>
  );
};
ActionSheetBase.Item = Item;
ActionSheetBase.FirstItem = FirstItem;
ActionSheetBase.InnerItem = InnerItem;
ActionSheetBase.LastItem = LastItem;
ActionSheetBase.CancelItem = CancelItem;
ActionSheetBase.Label = Label;

export default ActionSheetBase;
