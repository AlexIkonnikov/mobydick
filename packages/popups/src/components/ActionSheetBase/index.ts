import ActionSheetBase from './ActionSheetBase';

export {ActionSheetBase};
