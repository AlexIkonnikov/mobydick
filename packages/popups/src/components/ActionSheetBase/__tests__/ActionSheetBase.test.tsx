import {render} from '@testing-library/react-native';
import React from 'react';

import ActionSheetBase from '../ActionSheetBase';

describe('@npm/mobydick-popups/ActionSheetBase', () => {
  it('should renders correctly', () => {
    const {toJSON} = render(<ActionSheetBase id={'id'} onClose={() => null} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
