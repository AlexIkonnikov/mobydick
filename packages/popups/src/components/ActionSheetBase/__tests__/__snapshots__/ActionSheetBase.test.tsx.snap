// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/ActionSheetBase should renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "justifyContent": "flex-end",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="Pressable"
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`@npm/mobydick-popups/ActionSheetBase should renders correctly with item 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "justifyContent": "flex-end",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="Pressable"
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityLabel="accessibilityLabelActionSheetsItem"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderTopColor": "#D1D4DB",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 12,
            "minHeight": 75,
            "paddingHorizontal": 30,
          },
          Object {
            "backgroundColor": "#FFF",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontFamily": "Inter-Regular",
                "fontSize": 22.5,
                "lineHeight": 28.5,
              },
              Object {
                "paddingVertical": 22.5,
              },
            ]
          }
        >
          title
        </Text>
      </View>
    </View>
  </View>
</View>
`;
